python/gtsam/examples/SFMExample_bal.py

```diff
@@ -7,49 +7,58 @@
   See LICENSE for the license information
 
   Solve a structure-from-motion problem from a "Bundle Adjustment in the Large" file
-  Author: Frank Dellaert (Python: Akshay Krishnan, John Lambert)
+  Author: Frank Dellaert (Python: Akshay Krishnan, John Lambert, Varun Agrawal)
 """
 
 import argparse
 import logging
 import sys
 
-import matplotlib.pyplot as plt
-import numpy as np
-
 import gtsam
-from gtsam import (
-    GeneralSFMFactorCal3Bundler,
-    PinholeCameraCal3Bundler,
-    PriorFactorPinholeCameraCal3Bundler,
-    readBal,
-    symbol_shorthand
-)
+from gtsam import (GeneralSFMFactorCal3Bundler,
+                   PriorFactorPinholeCameraCal3Bundler, PriorFactorPoint3,
+                   readBal)
+from gtsam.symbol_shorthand import C, P
+from gtsam.utils.plot import plot_3d_points, plot_trajectory
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
 
-C = symbol_shorthand.C
-P = symbol_shorthand.P
 
+def plot(scene_data: gtsam.SfmData, result: gtsam.Values):
+    """Plot the trajectory."""
+    plot_vals = gtsam.Values()
+    for cam_idx in range(scene_data.number_cameras()):
+        plot_vals.insert(C(cam_idx),
+                         result.atPinholeCameraCal3Bundler(C(cam_idx)).pose())
+    for t_idx in range(scene_data.number_tracks()):
+        plot_vals.insert(P(t_idx), result.atPoint3(P(t_idx)))
 
-logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+    plot_3d_points(0, plot_vals, linespec="g.")
+    plot_trajectory(0, plot_vals, show=True)
 
-def run(args):
+
+def run(args: argparse.Namespace):
     """ Run LM optimization with BAL input data and report resulting error """
-    input_file = gtsam.findExampleDataFile(args.input_file)
+    if args.input_file:
```

Comment:
I think it might be better for this function to check that args.input_file exists and can be read. If not, it can fail. 
The default is probably better provided as a flag default. Are there other reasons why you prefer it this way?

Reply:
`readBal` would fail if the file didn't exist so I am not sure what the ask is here.

Reply:
I updated the default flag in a way that should make you happy. :slightly_smiling_face: